modules/openxr/extensions/openxr_fb_foveation_extension.h

```diff
@@ -82,7 +82,15 @@ class OpenXRFBFoveationExtension : public OpenXRExtensionWrapper {
 	XrFoveationLevelFB foveation_level = XR_FOVEATION_LEVEL_NONE_FB;
 	XrFoveationDynamicFB foveation_dynamic = XR_FOVEATION_DYNAMIC_DISABLED_FB;
 
-	void update_profile();
+	static void _update_profile();
+
+	void update_profile() {
+		// If we're rendering on a separate thread, we may still be processing the last frame, don't communicate this till we're ready...
+		RenderingServer *rendering_server = RenderingServer::get_singleton();
+		ERR_FAIL_NULL(rendering_server);
+
+		rendering_server->call_on_render_thread(callable_mp_static(&OpenXRFBFoveationExtension::_update_profile));
```

Comment:
Why `callable_mp_static()` and not `callable_mp()` so that `_update_profile()` doesn't need to grab the singleton instance and all that?

Reply:
Not sure, that didn't work when I was applying this for OpenXRAPI, but that might have been related to the parameters we pass there, and I just stuck to that approach :). I'll give it a try.

Reply:
Ah no, I remember now, `callable_mp` only works for objects subclassed from `Object`. We didn't do that for `OpenXRExtensionWrapper`, same as `OpenXRAPI` as they are internal classes not exposed to GDScript and to keep them light.

Reply:
Ah, ok, makes sense!